src/deployment/deploy.py

```diff
@@ -658,19 +659,33 @@ def deploy_app(self):
         with tempfile.TemporaryDirectory() as tmpdirname:
             with zipfile.ZipFile(self.app_zip, "r") as zip_ref:
                 zip_ref.extractall(tmpdirname)
-                subprocess.check_output(
-                    [
-                        shutil.which("func"),
-                        "azure",
-                        "functionapp",
-                        "publish",
-                        self.application_name,
-                        "--python",
-                        "--no-build",
-                    ],
-                    env=dict(os.environ, CLI_DEBUG="1"),
-                    cwd=tmpdirname,
-                )
+                error: Optional[subprocess.CalledProcessError] = None
+                max_tries = 5
+                for i in range(max_tries):
+                    try:
+                        subprocess.check_output(
+                            [
+                                shutil.which("func"),
+                                "azure",
+                                "functionapp",
+                                "publish",
+                                self.application_name,
+                                "--python",
+                                "--no-build",
+                            ],
+                            env=dict(os.environ, CLI_DEBUG="1"),
+                            cwd=tmpdirname,
+                        )
+                        return
+                    except subprocess.CalledProcessError as err:
+                        error = err
```

Comment:
Can we log the error here. So we have a record of all the errors in case it is failing for different reasons

Reply:
I will update this to add it to debug logging, but I'd prefer not to include it by default.  The logs are extremely verbose.  I'd prefer that the output be as simple as possible during expected operations.